src/query/tests/queryFn.test.tsx

```diff
@@ -247,7 +247,7 @@ describe('queryFn base implementation tests', () => {
     {
       const thunk = mutationWithNeither.initiate('mutationWithNeither')
       const result = await store.dispatch(thunk)
-      expect(result.error).toEqual(
+      expect((result as any).error).toEqual(
```

Comment:
FYI this change is because `result` previously got inferred as `any`, but is now more accurately:

```ts
{ data: string; } | { error: string | SerializedError }
```

If anyone has a better idea for handling this than `any-casting` I'm all ears

Reply:
Decided on a better option than `as any`:
```ts
      expect('error' in result && result.error).toEqual(
```